### PR TITLE
Feature: Dropdown options with callback

### PR DESCRIPTION
**Pull request by jeptechnology**
I wanted two things on the dropdown in my application: 
- to pass in an "on changed" callback to a dropdown. 
- to select whether to draw a border.

Can you consider this PR, please?

Thanks 👍

### CODE REVIEW

**Review by ArthurSonzogni — COMMENTED** [review]
Thanks!

File: examples/component/dropdown.cpp [hunk 1]
```diff
@@ -30,12 +30,12 @@ int main() {
 
   auto layout = Container::Vertical({
       Container::Horizontal({
-          Dropdown(&entries, &selected_1),
+          Dropdown(&entries, &selected_1, { .on_change = [&] { selected_3 = selected_1; } } ), // Keep in sync with selected_3.
```
[the remark this answers is not in the document]
Comment (jeptechnology) [reply examples/component/dropdown.cpp]: Sure - done 👍

File: include/ftxui/component/component.hpp [hunk 2]
```diff
@@ -74,7 +74,7 @@ Component Radiobox(ConstStringListRef entries,
                    int* selected_,
                    RadioboxOption options = {});
 
-Component Dropdown(ConstStringListRef entries, int* selected);
+Component Dropdown(ConstStringListRef entries, int* selected, DropdownOption options = {});
```
[the remark this answers is not in the document]
Comment (jeptechnology) [reply include/ftxui/component/component.hpp]: Done 👍

**Comment by jeptechnology** [comment]
@ArthurSonzogni Thanks for enhancing the dropdown functionality. I hope you don't mind but I saw the build error and put in a fix to correct it. Let me know if you think this is OK 👍

**Comment by ArthurSonzogni** [comment]
Thanks for your contribution to FTXUI! :+1:

**Pull request merged**